ambry-store/src/test/java/com.github.ambry.store/StoreMessageReadSetTest.java

```diff
@@ -290,18 +290,22 @@ private void verifyGetters(BlobReadOptions options, LogSegment logSegment, long
       MessageInfo msgInfo) {
     assertEquals("LogSegment name not as expected", logSegment.getName(), options.getLogSegmentName());
     assertEquals("Offset not as expected", offset, options.getOffset());
-    assertEquals("Size not as expected", msgInfo.getSize(), options.getSize());
-    assertEquals("ExpiresAtMs not as expected", msgInfo.getExpirationTimeInMs(), options.getExpiresAtMs());
-    assertEquals("StoreKey not as expected", msgInfo.getStoreKey(), options.getStoreKey());
+    assertEquals("Size not as expected", msgInfo.getSize(), options.getMessageInfo().getSize());
+    assertEquals("ExpiresAtMs not as expected", msgInfo.getExpirationTimeInMs(),
+        options.getMessageInfo().getExpirationTimeInMs());
+    assertEquals("StoreKey not as expected", msgInfo.getStoreKey(), options.getMessageInfo().getStoreKey());
     if (verifyInMemFields) {
-      assertEquals("Crc not as expected ", msgInfo.getCrc().longValue(), options.getCrc());
-      assertEquals("AccountId not as expected", msgInfo.getAccountId(), options.getAccountId());
-      assertEquals("ContainerId not as expected", msgInfo.getContainerId(), options.getContainerId());
-      assertEquals("OperationTimeMs not as expected", msgInfo.getOperationTimeMs(), options.getOperationTimeMs());
+      assertEquals("Crc not as expected ", msgInfo.getCrc().longValue(), options.getMessageInfo().getCrc().longValue());
```

Comment:
not really. I did thought about it too. At some places, we expected default accountId and containerId and nulls for CRCs(in memory vs not), which means we have to generate a new MessageInfo so that we can compare it with the actual. Also, we can't have these fields in equals in such cases. Efforts being the same is my opinion.